apluslms_roman/observer.py

```diff
@@ -151,38 +154,86 @@ def container_msg(self, step, msg):
         self._send_message(Message.CONTAINER_MSG, step, msg)
 
     def result_msg(self, result):
+        state = (StepState.SUCCEEDED if result.ok
+            else (StepState.CANCELLED if result.cancelled
+                else StepState.FAILED))
+        # NOTE: if for some reason some step isn't marked as succeeded,
+        # this will result in the 'ok: [time]' message, without any
+        # indication what step is in question and with an incorrect time
+        if state != StepState.SUCCEEDED:
+            self._state_update(result.step, state)
```

Comment:
The current logic, doesn't update earlier steps... I think that part is missing.

Reply:
...right I stashed that and then forgot it